app/components/dashboard/hypertension/patients_protected_component.rb

```diff
@@ -1,16 +1,15 @@
 class Dashboard::Hypertension::PatientsProtectedComponent < ApplicationComponent
-  attr_reader :data, :contactable, :period, :period_start
+  attr_reader :period, :period_start
 
-  def initialize(region:, data:, period:)
+  def initialize(region:, period:)
     @region = region
-    @data = data
     # Show only data up to last month
     @period = period.advance(months: -1)
     @period_start = period.advance(months: -36)
+    @repo = Reports::Repository.new(@region, periods: Range.new(@period_start, @period))
   end
 
   def graph_data
-    patients_protected = fill_empty_period_data(:controlled_patients)
     {
       patientsProtected: patients_protected,
       patientsProtectedWithSuffix: patients_protected.map { |k, v| [k, "#{v} patients"] }.to_h
```

Comment:
[nit-pick]: you could #transform_values instead of #map
```suggestion
      patientsProtectedWithSuffix: patients_protected.transform_values { |v| "#{v} patients" }
```